packages/react/src/components/header/components/headerNavigationMenu/HeaderNavigationMenu.tsx

```diff
@@ -25,7 +25,7 @@ export type HeaderNavigationMenuProps = React.PropsWithChildren<{
 
 export const HeaderNavigationMenu = ({ ariaLabel, children, id }: HeaderNavigationMenuProps) => {
   const { isSmallScreen } = useContext(HeaderContext);
-  const [openIndex, setOpenIndex] = useState<null | string>(null);
+  const [openIndex, setOpenIndex] = useState<undefined | string>(undefined);
```

Comment:
How about we made a global type for values that can get undefined as their values? As in
`type Unsure<XD> = XD | undefined`?

Would that be helpful? Could have types for nullable values as well etc.

Reply:
That's a good idea! 💡

Reply:
Is there a reason all indexes are strings and not numbers?

Reply:
Well, there was a reason before but it's not valid anymore. The indexes are not compared to anything so I could change them to numbers. That way they're defined more easily and typo-free.

Reply:
Good! Then there is no need for undefined/null as "-1" is common way to indicate index is not set.

Reply:
Oh man, I got really excited about @tuomashatakka 's idea.. 😆 But I think this is a simpler fix.